spec/system/nested_documentable_spec.rb

```diff
@@ -0,0 +1,270 @@
+
+        expect(page).to have_css ".file-name", text: "empty.pdf"
+      end
+
+      scenario "Should update nested document file title with
+                file name after choosing a file when no title defined" do
+        documentable_attach_new_file(file_fixture("empty.pdf"))
+
+        expect(page).to have_field("Title", with: "empty.pdf")
+      end
+
+      scenario "Should not update nested document file title with
+                file name after choosing a file when title already defined" do
+        click_link "Add new document"
+        within "#nested-documents" do
+          fill_in "Title", with: "My Title"
+          attach_file "Choose document", file_fixture("empty.pdf")
+        end
+
+        expect(page).to have_field("Title", with: "My Title")
+      end
+
+      scenario "Should update document cached_attachment field after valid file upload" do
+        click_link "Add new document"
+
+        cached_attachment_field = find("input[name$='[cached_attachment]']", visible: :hidden)
+        expect(cached_attachment_field.value).to be_empty
+
+        attach_file "Choose document", file_fixture("empty.pdf")
+
+        expect(page).to have_css(".loading-bar.complete")
+        expect(cached_attachment_field.value).not_to be_empty
+      end
+
+      scenario "Should not update document cached_attachment field after invalid file upload" do
+        documentable_attach_new_file(file_fixture("logo_header.gif"), false)
+
+        cached_attachment_field = find("input[name$='[cached_attachment]']", visible: :hidden)
+        expect(cached_attachment_field.value).to be_empty
+      end
+
+      scenario "Should show document errors after documentable submit with empty document fields" do
+        click_link "Add new document"
+        click_button submit_button_text
+
+        within "#nested-documents .document-fields" do
+          expect(page).to have_content("can't be blank", count: 2)
+        end
+      end
+
+      scenario "Should delete document after valid file upload and click on remove button" do
+        documentable_attach_new_file(file_fixture("empty.pdf"))
+        click_link "Remove document"
+
+        expect(page).not_to have_css("#nested-documents .document-fields")
+      end
+
+      scenario "Should show successful notice when resource filled correctly without any nested documents" do
+        fill_in_required_fields(factory, path)
+        click_button submit_button_text
+
+        expect(page).to have_content notice_text
+      end
+
+      scenario "Should show successful notice when resource filled correctly and after valid file uploads" do
+        fill_in_required_fields(factory, path)
+
+        documentable_attach_new_file(file_fixture("empty.pdf"))
+        click_button submit_button_text
+
+        if factory != :dashboard_action
+          within "#documents" do
+            expect(page).to have_content "Documents (1)"
+            expect(page).to have_css "a[rel=nofollow]", text: "empty.pdf"
```

Comment:
We used to check that the link pointed to a URL ending with `.pdf`, but we're no longer doing so :thinking:. Should we add the expectation back?

Reply:
I see now there's a line saying: `expect(page).to have_link text: "empty.pdf"` :thinking:. This checks the text of the link, but previously we tested the URL :thinking:.